service-clients-core/src/main/java/com/flipkart/poseidon/serviceclients/ServiceResponse.java

```diff
@@ -43,6 +48,15 @@ public T getData() {
         return data;
     }
 
+    public void addData(T data) {
+        if (data != null)
+            dataList.add(data);
+    }
```

Comment:
I think we shouldn't have null check for data here. Response list size should match request list size always. Even before this change, if actual data is null, we still return it to callers. You may have to add null also to list.